### PR TITLE
WIP: task-NXP-27503-NXP-27504-platform-simple-pipeline

### PR DESCRIPTION
**Pull request by ataillefer**
Pipeline running in the "platform" team's Jenkins X to run the Maven build + unit tests on the repository: http://jenkins.platform.34.74.59.50.nip.io/job/nuxeo/job/nuxeo/view/change-requests/job/PR-3150/

To be reviewed but **cannot be merged for now**, waiting for:
- [NXP-27562](https://jira.nuxeo.com/browse/NXP-27562): MongoDB tests should be ignored when running unit tests by default
- [NXP-27324](https://jira.nuxeo.com/browse/NXP-27324): Move all JSF UI code into its own nuxeo-jsf-ui plugin
- A certain number of issues on unit tests failing due to converter issues that need to be investigated.

Once these issues are solved/merged, the temporary commits will be removed.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+  stages {
+    stage('Compile, package and install') {
+      steps {
+        container('maven') {
+          echo """
+          ----------------------------------------
+          Compile, package and install
+          ----------------------------------------"""
+          withEnv(["MAVEN_OPTS=$MAVEN_OPTS -Xms3072m -Xmx3072m"]) {
+            echo "MAVEN_OPTS=$MAVEN_OPTS"
```
Comment (kevinleturc) [inline Jenkinsfile]: Forgotten debug message?

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        }
+        failure {
+          setGitHubBuildStatus('utests/dev', 'Unit tests - dev environment', 'FAILURE')
+        }
+      }
+    }
+  }
+  post {
+    always {
+      junit testResults: '**/target/surefire-reports/*.xml'
```
Comment (kevinleturc) [inline Jenkinsfile]: I think `junit` crashes if the `testResults` are empty, which can be the case if compile crashes. Shouldn't we put it in the post/always of unit tests stage?

Does the failsafe reports has to be taken into account in this PR? there might be some run.

File: Jenkinsfile (hunk 1 above)
Comment (ataillefer) [reply Jenkinsfile]: Nope, wanted to have them in the logs.

File: Jenkinsfile (hunk 2 above)
Comment (ataillefer) [reply Jenkinsfile]: Yes about `junit` in the unit tests stage.
There are no tests run by failsafe in this build.

**Comment by ataillefer** [comment]
> Great works! Want to use it asap!
Thanks! Not sure when though :smiley: 

Yes I'll add the pending status at the beginning of each stage.

**Comment by ataillefer** [comment]
Closed in favor of https://github.com/nuxeo/nuxeo/pull/3183.

**Pull request closed without merging**